src/test/storage/btree_index_test.cpp

```diff
@@ -69,7 +69,26 @@ TEST_F(BTreeIndexTest, IndexProbes) {
 // Short String Optimization (SSO) stores strings of a certain size in the pmr_string object itself.
 // Only strings exceeding this size (15 for libstdc++ and 22 for libc++) are stored on the heap.
 
-TEST_F(BTreeIndexTest, MemoryConsumptionVeryShortString) {
+/*
+  size_t memory_consumption() const;
+  
+  |    Characteristic               | Block 1 | Block 2 |
+  |---------------------------------|---------|---------|
+  |[A] index is empty               |    true |   false |
+  |[B] index has null positions     |    true |   false |
+  |[C] index has non-null positions |    true |   false |
```

Comment:
How to read this table and the choices / combinations below?

Reply:
Input Domain Modeling is used here as testing technique. The input domain is scoped by parameters, which are structured in terms of characteristics (here `index is empty`, for example).
These characteristics are then partitioned into blocks, which represent the different options for a characteristic and thus sets of values for the actual test case. After characteristics are defined and partitioned into blocks, combinations of these blocks have to be chosen. Choosing all combinations is usually infeasible. (Since only three characteristics are listed with only two blocks each, choosing all combinations would be possible here.) Instead of choosing all combinations, other choosing strategies such as _Each Choice_, _Pair-Wise Choice_ and _Base Choice_ exist.

The added tests are created by using the _Base Choice_:
>Base Choice Coverage (BCC): A base choice block is chosen for each characteristic, and a base test is formed by using the base choice for each characteristic. Subsequent tests are chosen by holding all but one base choice constant and using each non-base choice in each other characteristic.
[AmOf17]

Therefore, a specific base choice has to be chosen. I decided to use the most likely combination from an end-use point of view, which is (a) the index is not empty, (b) the index stores NULL-positions and (c) the index stores non-NULL-positions. With this base choice, the additional combinations to be tested can be calculated as mentioned in the quote.

Base choice:
  A2, B1, C1
Hold [A] and [B] constant but use all other [C]-blocks yields to:
  A2, B1, C2
Hold [A] and [C] constant but use all other [B]-blocks yields to:
  A2, B2, C1
Hold [B] and [C] constant but use all other [B]-blocks yields to:
  A1, B1, C1
Since this derived combination is infeasible (index is empty but contains postings), one of the blocks held constant is changed so that a feasible combination is achieved:
A1, B2, C2

[AmOf17] Introduction to Software Testing (2nd Edition); Paul Ammann, Jeff Offutt, Cambridge University Press, 2017

Reply:
Never heard of it. Choosing one default configuration and varying one parameter after another is common, but I did not know that it has a name. For more complex tests, those naming schemes make sense. However, I expect that I am not the only one unfamiliar with them. Here, it might be sufficient to write `non-empty index covering both NULL and non-NULL position` instead of `A2, B1, C1`. That should make it easier for the reader.

Reply:
Just saw that the combinations in the other test are more complex. I would still suggest auto-replacing them with the textual description (or at least adding a description). If you leave the scheme in, please add a reference to a short description of the testing pattern.